core/controller/src/main/scala/whisk/core/loadBalancer/InvokerSupervision.scala

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright 2015-2016 IBM Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package whisk.core.loadBalancer
+
+import scala.concurrent.duration._
+import scala.concurrent.Future
+
+import akka.actor.Actor
+import akka.actor.FSM
+import akka.actor.Props
+import akka.pattern.ask
+import akka.pattern.pipe
+import akka.util.Timeout
+import akka.actor.FSM.SubscribeTransitionCallBack
+import akka.actor.FSM.Transition
+import whisk.common.AkkaLogging
+import whisk.common.TransactionId
+import whisk.core.connector.PingMessage
+import whisk.common.KeyValueStore
+import spray.json._
+import spray.json.DefaultJsonProtocol._
+import whisk.common.ConsulKV.LoadBalancerKeys
+
+// Received events
```

Comment:
should there be any package protection on these types? do they need to be global?